temporal-sdk/src/main/java/io/temporal/activity/ActivityExecutionContext.java

```diff
@@ -140,4 +141,10 @@ public interface ActivityExecutionContext {
    * WorkflowServiceStubsOptions.Builder#setMetricsScope(Scope)} when a worker starts up.
    */
   Scope getMetricsScope();
+
+  /**
+   * Get a {@link WorkflowClient} that can be used to start interact with the Temporal service from
+   * an activity.
+   */
+  WorkflowClient getWorkflowClient();
```

Comment:
I assume we don't expect users to have their own implementations of `ActivityExecutionContext`? I know we have a `TestActivityEnvironment`, just wanted to check. If you do make another commit in this PR, may be worth a warning at the top of this interface saying that methods may be added without default implementations and that users should not implement this interface.

Reply:
We have `ActivityExecutionContextBase` if users want to provide their own wrapper implementation but not have to implement every method.